x/core/types/msgs.go

```diff
@@ -0,0 +1,283 @@
+		return ErrInvalidLengthExecProgramID.Wrapf("given ID is %d characters long", len(m.ExecProgramId))
+	}
+	if len(m.TallyProgramId) != 64 {
+		return ErrInvalidLengthTallyProgramID.Wrapf("given ID is %d characters long", len(m.TallyProgramId))
+	}
+
+	if !semver.IsValid("v"+m.Version) || semver.Prerelease("v"+m.Version) != "" || semver.Build("v"+m.Version) != "" {
+		return ErrInvalidVersion.Wrapf("%s", m.Version)
+	}
+
+	if len(m.ExecInputs) > int(config.ExecInputLimitInBytes) {
+		return ErrExecInputLimitExceeded.Wrapf("%d bytes > %d bytes", len(m.ExecInputs), config.ExecInputLimitInBytes)
+	}
+	if len(m.TallyInputs) > int(config.TallyInputLimitInBytes) {
+		return ErrTallyInputLimitExceeded.Wrapf("%d bytes > %d bytes", len(m.TallyInputs), config.TallyInputLimitInBytes)
+	}
+	if len(m.ConsensusFilter) > int(config.ConsensusFilterLimitInBytes) {
+		return ErrConsensusFilterLimitExceeded.Wrapf("%d bytes > %d bytes", len(m.ConsensusFilter), config.ConsensusFilterLimitInBytes)
+	}
+	if len(m.Memo) > int(config.MemoLimitInBytes) {
+		return ErrMemoLimitExceeded.Wrapf("%d bytes > %d bytes", len(m.Memo), config.MemoLimitInBytes)
+	}
+	if len(m.PaybackAddress) > int(config.PaybackAddressLimitInBytes) {
+		return ErrPaybackAddressLimitExceeded.Wrapf("%d bytes > %d bytes", len(m.PaybackAddress), config.PaybackAddressLimitInBytes)
+	}
+	if len(m.SEDAPayload) > int(config.SEDAPayloadLimitInBytes) {
+		return ErrSEDAPayloadLimitExceeded.Wrapf("%d bytes > %d bytes", len(m.SEDAPayload), config.SEDAPayloadLimitInBytes)
+	}
+
+	return nil
+}
+
+// MsgHash returns the hex-encoded hash of the PostDataRequest message to be used
+// as the data request ID.
+func (m *MsgPostDataRequest) MsgHash() (string, error) {
+	execProgramIDBytes, err := hex.DecodeString(m.ExecProgramId)
+	if err != nil {
+		return "", err
+	}
+	tallyProgramIDBytes, err := hex.DecodeString(m.TallyProgramId)
+	if err != nil {
+		return "", err
+	}
+
+	execInputsHasher := sha3.NewLegacyKeccak256()
+	execInputsHasher.Write(m.ExecInputs)
+	execInputsHash := execInputsHasher.Sum(nil)
+
+	tallyInputsHasher := sha3.NewLegacyKeccak256()
+	tallyInputsHasher.Write(m.TallyInputs)
+	tallyInputsHash := tallyInputsHasher.Sum(nil)
+
+	consensusFilterHasher := sha3.NewLegacyKeccak256()
+	consensusFilterHasher.Write(m.ConsensusFilter)
+	consensusFilterHash := consensusFilterHasher.Sum(nil)
+
+	memoHasher := sha3.NewLegacyKeccak256()
+	memoHasher.Write(m.Memo)
+	memoHash := memoHasher.Sum(nil)
+
+	execGasLimitBytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(execGasLimitBytes, m.ExecGasLimit)
+	tallyGasLimitBytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(tallyGasLimitBytes, m.TallyGasLimit)
+	replicationFactorBytes := make([]byte, 2)
+	//nolint:gosec // G115: No overflow guaranteed by validation logic.
+	binary.BigEndian.PutUint16(replicationFactorBytes, uint16(m.ReplicationFactor))
+
+	dataRequestHasher := sha3.NewLegacyKeccak256()
+	dataRequestHasher.Write([]byte(m.Version))
```

Comment:
Should be the hash of the version string as bytes. So `hash([]byte(m.Version))`

Reply:
Good catch, thanks. Not sure if we need this field from now on but I will address in the new PR.